### PR TITLE
Upgrade to es2015 - issue 334

### PR DESCRIPTION
**Pull request by mario-git**
https://github.com/bbyars/mountebank/issues/334

I know this is a massive refactoring, I did this constantly running the tests and changing only on the test side or on the code base side to avoid surprises. I know the project contributing guideline suggests "the smaller [pull request] the better". If this is too big (it probably is) I will take this and batch it in smaller chunks.

Anyway, what changed? In short:

- I introduced const/let where appropriate;
- same for arrow functions;
- same for string template literals (I am sure there are still some string concatenation in the code base, but I changed plenty of them);
- removed "this" in tests (refactored this.timeout; removed this.name for requestName, it shouldn't be a huge problem);

(- spread operator, one of the coolest features... I think I haven't found any place where to use it, mountebank can live without that for the time being!)

I didn't introduce arrow functions where it could bring trouble, mainly where "this" is used quite a lot, for example src/views folder (one day this will change entirely when jQuery will be removed, see issue https://github.com/bbyars/mountebank/issues/348, tasks folders and some utils file).

I changed a lint rule (no-confusing-arrow, to allow ternary in arrow functions). Initially to make lint happy I was changing the code base, then I changed my mind because I fundamentally disagreed with this lint rule.

Promises with Q are still there, a transition to native promises probably needs to be done separately.

### CODE REVIEW

**Comment by mario-git** [comment]
I know there's a problem with one of the non-airplane tests and clearly the first/easy attempt to fix that failed, I'll have a deeper look as soon as I can.

File: functionalTest/api/foo/fooImposterTest.js [hunk 1]
```diff
@@ -1,78 +1,62 @@
...
 
-var assert = require('assert'),
+const assert = require('assert'),
     api = require('../api').create(),
     tcp = require('../tcp/tcpClient'),
     promiseIt = require('../../testHelpers').promiseIt,
     port = api.port + 1,
-    timeout = parseInt(process.env.MB_SLOW_TEST_TIMEOUT || 2000);
+    timeout = parseInt(process.env.MB_SLOW_TEST_TIMEOUT || 2000),
+    requestName = 'some request name';
```
Comment (bbyars) [inline functionalTest/api/foo/fooImposterTest.js]: This loses the original context (the "this.name" was passed so you could see the test name in the logs). It's not really important to keep it; probably the best thing is just to drop the name field in all the requests

File: functionalTest/api/foo/fooStubTest.js [hunk 2]
```diff
@@ -1,107 +1,90 @@
...
-            var originServerPort = port + 1,
+        promiseIt('should allow proxy stubs', () => {
+            const originServerPort = port + 1,
                 originServerStub = { responses: [{ is: { data: 'PROXIED' } }] },
                 originServerRequest = {
                     protocol: 'foo',
                     port: originServerPort,
                     stubs: [originServerStub],
-                    name: this.name + ' ORIGIN'
+                    name: `${requestName} ORIGIN`
```
Comment (bbyars) [inline functionalTest/api/foo/fooStubTest.js]: This is an example where, if we can't get the name of the test ("should allow proxy stubs") or it's hard, we should just remove requestName. As is, it's more confusing than just leaving it as "ORIGIN"

**Comment by bbyars** [comment]
Thanks again!

**Pull request merged**